### PR TITLE
Refactor JSON-LD serialization for CodeClimate

### PR DESCRIPTION
**Pull request by brunnohenrique**
## What type of PR is this? (check all applicable)

- [X] Refactor
- [ ] Feature
- [ ] Bug Fix
- [ ] Optimization
- [ ] Documentation Update

## Description
Move methods at articles_controller.rb to a new service that creates json.

## Related Tickets & Documents
https://github.com/forem/forem/issues/3739

## Added tests?

- [X] yes
- [ ] no, because they aren't needed
- [ ] no, because I need help

## Added to documentation?

- [ ] docs.forem.com
- [ ] readme
- [X] no documentation needed

## [optional] What gif best describes this PR or how it makes you feel?

alt_text

### CODE REVIEW

File: app/controllers/stories_controller.rb [hunk 1]
```diff
@@ -200,7 +200,7 @@ def handle_organization_index
       .limited_column_select
       .order(published_at: :desc).page(@page).per(8))
     @organization_article_index = true
-    set_organization_json_ld
+    @organization_json_ld = Articles::JsonCreator.organization(@organization)
```
Comment (rhymes) [inline app/controllers/stories_controller.rb:203]: Hi @brunnohenrique, I'm not convinced this is the right solution. This is not about creating JSON per se but it's just a different representation of the org object, or user object or article object. I think we should respect that and use serializers, similar to `app/serializers`

In this case you're serializing `JSON-LD`, so I would imagine something like `app/serializers/json_ld/article_serializer.rb` with the same structure of the existing ones. Same for the other models

cc @citizen428

**Review by rhymes — COMMENTED** [review]
Thanks for tackling this @brunnohenrique, left some feedback

File: app/controllers/stories_controller.rb (hunk 1 above)
Comment (citizen428) [reply app/controllers/stories_controller.rb:203]: Yes, I recently mentioned this in another PR too (can't remember which one though): if the main purpose of a service is generating a JSON response, it's a serializer in disugise. 😉

**Comment by rhymes** [comment]
Hi @brunnohenrique, let us know if you need any help carrying this to the finish line. Thank you for your contribution!

**Comment by brunnohenrique** [comment]
Hi @rhymes !  
I was too busy in the lasts weeks, but now I'm back! I'll try to deliver this PR until this weekend.

**Comment by msarit** [comment]
Hey @brunnohenrique! 👋🏾
Just drawing your attention to this PR; some feedback has been left for you.
Let us know if you need help or have any questions.

**Comment by aitchiss** [comment]
Hey @brunnohenrique 

Thank you so much for your contribution! Since this PR has not been active for a while I am going to go ahead and close it. If you find the time to make the requested changes & resolve the conflicts, please feel free to reopen and re-request a review from our team!

We couldn't do what we do without awesome contributors like yourself, thank you so much for being a part of our awesome Forem community 🤗

**Pull request closed without merging**